--- unification-fd-0.11.2/unification-fd.cabal	2001-09-08 21:46:40.000000000 -0400
+++ unification-fd-0.11.2/unification-fd.cabal	2024-05-21 17:53:32.740519714 -0400
@@ -70,7 +70,7 @@
                    -- should be resolved now. Cf.,
                    -- <https://github.com/Bodigrim/logict/issues/20#issuecomment-774528439>
                    -- <https://github.com/wrengr/unification-fd/issues/14>
-    Build-Depends:   logict       >= 0.4 && < 0.8.1
+    Build-Depends:   logict       >= 0.4 && < 0.8.2
                    -- N.B., Tasty requires base>=4.5.0.0; which
                    -- means we aren't CI testing anything older than
                    -- that anymore, so we might as well just require
